### PR TITLE
Fix black test failure in CI 

### LINKED ISSUE #119: Failure of Black Test in CI testing in github
**Description of the bug**:
I opened a pull request to add some documentation files. However, during CI testing on GitHub, some .py files failed the black test (see image below and this [link](https://github.com/CosmoStat/wf-psf/actions/runs/7929828212/job/21650841152?pr=117)
 for the full log description), despite passing when I ran these tests on my Linux machine. Note that I didn't make any changes to these Python files in the PR.

pytest

**Expected behavior**:
 All files should pass the CI testing.

### PR DESCRIPTION
**Pull request by nadamoukaddem**
this PR should solve #119. 

To address the issue, I temporarily set the pytest version to 7.4.4 and reformatted the code. Additionally, I upgraded the Python version in CI to 3.10.12.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (jeipollack) [inline .github/workflows/ci.yml:1]: Why did you need to change the Python version?

Could you implement a matrix for the Python versions (e.g. 3.9, 3.10) following this [example](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#using-multiple-python-versions)?

File: pyproject.toml
Comment (jeipollack) [inline pyproject.toml:1]: I don't understand what you mean in the description that you "temporarily" set the pytest version to 7.4.4. 

Could you explain what you mean?

Comment (nadamoukaddem) [reply pyproject.toml:1]: Although the latest version (8.0.1) should be used, we've encountered error #118 for which we haven't yet found a solution. To ensure uninterrupted progress on my tasks, I'll temporarily use version 7.4.4 until we resolve the issue.

Comment (jeipollack) [reply pyproject.toml:1]: The version change is part of this Pull Request so it is not a temporary change.  Is this Pull Request ready for review or is it a draft? Could you also complete the task requested by Sam that I wrote in https://github.com/CosmoStat/wf-psf/issues/118#issuecomment-1950250511?

Do the comparison of versions of pytest-black in Pypi.org with that in the repository.  State whether the versions are consistent or not and clarify what is the main problem currently which prevents resolving the error reported in #118.

Comment (nadamoukaddem) [reply pyproject.toml:1]: yes the PR is ready for review.

Comment (jeipollack) [reply pyproject.toml:1]: Okay.  In order to complete this PR, I would like the answer to the questions and completion of tasks posed above.  
And, I would like for you to implement a working version of the matrix for different Python versions. If certain versions do not work state why and only add those which work.

Comment (nadamoukaddem) [reply pyproject.toml:1]: Could you please provide more context or clarify why it's necessary to test our code across multiple Python versions, especially the older ones. Understanding why this task is important will help me explain any issues with versions that don't work as expected.

Comment (jeipollack) [reply pyproject.toml:1]: It is good for users to know what versions WaveDiff is/is not compatible with.

Comment (nadamoukaddem) [reply pyproject.toml:1]: all checks have passed with 3.9 and 3.10

Comment (jeipollack) [reply pyproject.toml:1]: Okay, thanks for this update. 

Again, it's important for us to ensure that WaveDiff is compatible with various Python versions to cater to a broader user base. We need to consider that all users wish to be on the latest Python version (in contrast to us who are not), so by testing across multiple versions, we can guarantee a smoother experience for all users regardless of their Python environment.

Regarding the task at hand, you stated that all checks have passed with Python versions 3.9 and 3.10, which is great progress.  However, it is apparent you encountered failures with versions 3.11 and 3.12.

Given our use of Continuous Integration with GitHub Actions, it's essential to implement a matrix for launching unit tests across different Python versions in our CI workflow. This ensures that any changes introduced are thoroughly tested across various Python environments.

Here's a checklist to guide you further:

- [ ]  Implement a matrix in our GitHub Actions workflow YAML file to test the code with Python versions 3.9, 3.10, 3.11, and 3.12.

- [ ]  Ensure that the unit tests are launched for each Python version specified in the matrix.

- [ ]  Document any errors or failures encountered during testing with these versions, and open an issue for each encountered issue.

- [ ]  Update the matrix in the GitHub Actions workflow YAML file with the confirmed working versions.

Let's prioritize these steps for now, and we can address the broader compatibility issues in a separate Pull Request. If you have any questions or concerns about this approach, please feel free to discuss them here.

File: .github/pull_request_template.md
Comment (jeipollack) [inline .github/pull_request_template.md:1]: This new section is redundant and should instead be part of the code review process.  Delete.

At the bottom of the Reviewer Checklist, add


`- [ ] I have reviewed the CI log to verify if any files require formatting with Black. If any files are found to be improperly formatted, I have provided comments indicating the need for formatting.
`

File: pyproject.toml
Comment (jeipollack) [reply pyproject.toml:1]: Please add your comments here on what you changes you implemented. So, I can determine whether the conversation can be resolved.

File: .github/workflows/ci.yml
Comment (jeipollack) [reply .github/workflows/ci.yml:1]: Please note the changes you implemented here.

File: pyproject.toml
Comment (nadamoukaddem) [reply pyproject.toml:1]: Changes Made:
- Implemented a matrix for Python versions 3.9 and 3.10
- Removed pytest-black and pytest-cases
- Added the Black formatter to the CI pipeline
- Added a checkbox in the PR to prompt the reviewer to check the CI log for any required file changes.

File: pyproject.toml [hunk 1]
```diff
@@ -53,8 +53,8 @@ release = [
 
 test = [
   "pytest",
-  "pytest-black",
-  "pytest-cases",
+  #"pytest-black",
```
Comment (jeipollack) [inline pyproject.toml]: You commented them out, but we don't want commented lines in the `pyproject.toml` file.  Just remove them.

Comment (nadamoukaddem) [reply pyproject.toml]: ok I will. I commented them to remember to add them once the plugins are compatible with PyTest.

Comment (jeipollack) [reply pyproject.toml]: We should use issues as reminder.

Comment (nadamoukaddem) [reply pyproject.toml]: changes are done.
